collect_app/src/main/java/org/odk/collect/android/logic/FormController.java

```diff
@@ -109,7 +109,9 @@
 
         public InstanceMetadata(String instanceId, String instanceName, AuditConfig auditConfig) {
             this.instanceId = instanceId;
-            this.instanceName = instanceName;
+            this.instanceName = instanceName != null
+                    ? instanceName.replaceAll("[\\p{Cntrl}]", " ")
```

Comment:
Let's make `"[\\p{Cntrl}]"` a constant to DRY things up -- or even better make the whole thing a TextUtils function.